### ISSUE TITLE
How to use with less global variables

### ISSUE THREAD

**Issue by woshiguabi** [mrmckeb/typescript-plugin-css-modules#53]
**Describe the bug**
In my project, i use less-loader option `modifyVars` to support global variables.
But seems like `modifyVars` will cause plugin not working.

**To Reproduce**
_path/to/webpack.config.js_
``` 
{
  loader: 'less-loader',
  options: {
    modifyVars: require("path/to/theme.js")
  }
}
```
_path/to/theme.js_
```
{
  'global-variable': '#666'
}
```
_path/to/component.less_
```
.component {
  color: @global-variable;
}
```
_path/to/component.tsx_
```
import styles from 'path/to/component.less'; // empty object 
```

**Expected behavior**
The export object includes `component` prop.

**Desktop (please complete the following information):**

- OS: Win10
- Browser Chrome
- Version latest

**Additional context**
I know this plugin can't get less-loader global variables, so is there any solution for this issue?
Add an option to import the global variables may be a quick fix.

**Comment by pwang1984** [comment]
I have the same problem. In addition to the global variable, if I have any import to my less file, it stops working.

In my case, it could be something to do with visual studio code though. Everything compiles and runs properly. The problems are auto-completion not working and also visual studio code complaining (which is annoying)

**Comment by pwang1984** [comment]
logs:
```
Info 74   [12:56:28.318] FileWatcher:: Added:: WatchInfo: /home/pwang/dev/src/sirius/sirius-pos-ui/src/pages/splash/SplashPage.module.less 500 Project:  WatchType: Closed Script info
Info 75   [12:56:28.327] [typescript-plugin-css-modules] Failed TypeError: Cannot read property 'css' of undefined
Info 76   [12:56:28.337] [typescript-plugin-css-modules] Stack trace: TypeError: Cannot read property 'css' of undefined
    at /home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/lib/helpers/DtsSnapshotCreator.js:37:47
    at /home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:13037:18
    at /home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:13335:9
    at ImportVisitor.finish [as _finish] (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:8973:18)
    at ImportVisitor._onSequencerEmpty (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:6862:10)
    at ImportSequencer.tryRun (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:6824:14)
    at ImportVisitor.run (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:6855:21)
    at Object.parse (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:8978:53)
    at Object.parse (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:13330:50)
    at Object.render (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/node_modules/less/dist/less.cjs.js:13026:12)
    at DtsSnapshotCreator.getClasses (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/lib/helpers/DtsSnapshotCreator.js:36:22)
    at DtsSnapshotCreator.getDtsSnapshot (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/lib/helpers/DtsSnapshotCreator.js:87:28)
    at Object.ts.createLanguageServiceSourceFile (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript-plugin-css-modules/lib/index.js:63:53)
    at acquireOrUpdateDocument (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:104601:37)
    at Object.acquireDocumentWithKey (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:104574:20)
    at getOrCreateSourceFileByPath (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:125509:41)
    at Object.getOrCreateSourceFile [as getSourceFile] (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:125461:24)
    at findSourceFile (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:92864:29)
    at processImportedModules (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:93152:25)
    at findSourceFile (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:92917:17)
    at processImportedModules (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:93152:25)
    at findSourceFile (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:92917:17)
    at processImportedModules (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:93152:25)
    at findSourceFile (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:92917:17)
    at /home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:92763:85
    at getSourceFileFromReferenceWorker (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:92730:34)
    at processSourceFile (/home/pwang/dev/src/sirius/sirius-pos-ui/node_modules/typescript/lib/tsserver.js:92763:13)
```

With global variable.
```
LessError {message: "variable @margin-none is undefined", stack: undefined, type: "Name", filename: "./SplashPage.modules.less", index: 99, …}
```

With import
```
LessError {message: "Unrecognised input", stack: undefined, type: "Parse", filename: "/styleVariables.module.less", index: 0, …}
```

**Comment by pwang1984** [comment]
Did a bit more digging. global vars can be passed in as a option for render. Following works
```
less.render(
            css,
            {
                globalVars: {
                    "margin-none": "0px"
                },
                syncImport: true,
                filename: path.resolve(file)
            } as any,
            (err, output) => {
                console.log(err);
                console.log(output);
            },
        );
```
Maybe we can define a plugin option to load the varibles from a json file

For the my import problem, I saw this thread
https://github.com/less/less.js/issues/2342#issuecomment-67596931
Maybe all we need is to resolve the file path. 

With global variable, import will be less important.

**Comment by mrmckeb** [comment]
Hi @pwang1984, would it help if I added the ability to pass in options to `less.render`?

**Comment by mrmckeb** [comment]
Hi @BasilPP, we've just released v2.0.0 which enables passing through options to renderers.

Release notes and migration steps can be found here: https://github.com/mrmckeb/typescript-plugin-css-modules/releases/tag/v2.0.0

Please let me know if you have any other issues or feedback! And thanks for your patience.
